engine/src/state.rs

```diff
@@ -244,15 +210,13 @@ impl State {
     }
 
     pub fn play_turn(&mut self, piece: Piece, target_position: Position) -> Result<(), GameError> {
-        if self.board.piece_already_played(&piece) {
+        if self.board.piece_already_played(piece) {
             self.turn_move(piece, target_position)?
         } else {
             self.turn_spawn(piece, target_position)?
         }
-        self.update_history(&piece, &target_position);
-        self.update_hasher();
+        self.update_history(piece, target_position);
         self.next_turn();
-        self.shutout();
```

Comment:
I don't get where/how shutouts are handled now? What passes the turn back to the same player if opp is in shutout? next_turn only ever increments turn and switches color